src/commands/run.ts

```diff
@@ -405,18 +406,56 @@ const fluenceRun = async (args: RunArgs) => {
   }
 
   await initFluenceClient(args, args.maybeFluenceConfig);
-  const { Fluence, callAquaFunction } = await import("@fluencelabs/js-client");
+
+  const { Fluence, callAquaFunction, js2aqua, aqua2js } = await import(
+    "@fluencelabs/js-client"
+  );
+
+  const schema = functionCall.funcDef;
+
+  // TODO: remove this after DXJ-535 is done
+  const validatedRunData = Object.fromEntries(
+    Object.entries(args.runData ?? {}).map(([argName, argValue]) => {
+      assert(
+        typeof argValue !== "function",
+        "Should be impossible to pass function as an argument to fluence run",
+      );
+
+      const fields =
+        schema.arrow.domain.tag === "nil" ? {} : schema.arrow.domain.fields;
+
+      const argSchema = fields[argName];
+
+      assert(
+        argSchema !== undefined,
+        "Should be impossible because schema always contains same keys as runData",
+      );
+
+      assert(
+        argSchema.tag !== "arrow",
+        "Should be impossible to pass function as an argument to fluence run",
+      );
+
+      return [argName, js2aqua(argValue, argSchema, { path: [argName] })];
+    }),
+  );
 
   const result = await callAquaFunction({
-    args: args.runData ?? {},
+    script: functionCall.script,
     config: {},
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    def: functionCall.funcDef,
     peer: Fluence.getClient(),
-    script: functionCall.script,
+    args: validatedRunData,
   });
 
-  return result;
+  const returnSchema =
+    schema.arrow.codomain.tag === "unlabeledProduct" &&
+    schema.arrow.codomain.items.length === 1
+      ? schema.arrow.codomain.items[0]
+      : schema.arrow.codomain;
```

Comment:
can be done like this so there is no need for asserts
```suggestion
  const returnSchema =
    (schema.arrow.codomain.tag === "unlabeledProduct"
      ? schema.arrow.codomain.items[0]
      : undefined) ?? schema.arrow.codomain
```

Reply:
Won't work. It's different code. What if length property is `2`?

Reply:
yeah didn't correctly understand the code

Reply:
That's kinda of a problem that I don't really understand what is happening here, why you here care only about the case when there is 1 item

Reply:
```suggestion
const returnSchema =
    (schema.arrow.codomain.tag === "unlabeledProduct" &&
    schema.arrow.codomain.items.length === 1
      ? schema.arrow.codomain.items[0]
      : undefined) ?? schema.arrow.codomain;
```